draft-ietf-lamps-x509-slhdsa.md

```diff
@@ -297,187 +415,81 @@ TODO
-    PRIVATE-KEY SLH-DSA-PrivateKey }
-
-pk-slh-dsa-128f-sha2 PUBLIC-KEY ::= {
-    IDENTIFIER id-alg-slh-dsa-128f-sha2
-    KEY SLH-DSA-PublicKey
-    PARAMS ARE absent
-    CERT-KEY-USAGE
-        { digitalSignature, nonRepudiation, keyCertSign, cRLSign }
-    PRIVATE-KEY SLH-DSA-PrivateKey }
-
-pk-slh-dsa-192s-shake PUBLIC-KEY ::= {
-    IDENTIFIER id-alg-slh-dsa-192s-shake
-    KEY SLH-DSA-PublicKey
-    PARAMS ARE absent
-    CERT-KEY-USAGE
-        { digitalSignature, nonRepudiation, keyCertSign, cRLSign }
-    PRIVATE-KEY SLH-DSA-PrivateKey }
-
-pk-slh-dsa-192f-shake PUBLIC-KEY ::= {
-    IDENTIFIER id-alg-slh-dsa-192f-shake
-    KEY SLH-DSA-PublicKey
-    PARAMS ARE absent
-    CERT-KEY-USAGE
-        { digitalSignature, nonRepudiation, keyCertSign, cRLSign }
-    PRIVATE-KEY SLH-DSA-PrivateKey }
-
-pk-slh-dsa-256s-shake PUBLIC-KEY ::= {
-    IDENTIFIER id-alg-slh-dsa-256s-shake
-    KEY SLH-DSA-PublicKey
-    PARAMS ARE absent
-    CERT-KEY-USAGE
-        { digitalSignature, nonRepudiation, keyCertSign, cRLSign }
-    PRIVATE-KEY SLH-DSA-PrivateKey }
-
-pk-slh-dsa-256f-shake PUBLIC-KEY ::= {
-    IDENTIFIER id-alg-slh-dsa-256f-shake
-    KEY SLH-DSA-PublicKey
-    PARAMS ARE absent
-    CERT-KEY-USAGE
-        { digitalSignature, nonRepudiation, keyCertSign, cRLSign }
-    PRIVATE-KEY SLH-DSA-PrivateKey }
-
-SLH-DSA-PublicKey ::= OCTET STRING
-
-SLH-DSA-PrivateKey ::= OCTET STRING
-~~~~
-
 # Security Considerations
 
-The security considerations of [RFC5280] applies accordingly.
+The security considerations of {{RFC5280}} applies accordingly.
+
+Implementations MUST protect the private keys.  Compromise of the
+private keys may result in the ability to forge signatures.
+
+When generating an SLH-DSA key pair, an implementation MUST generate
+each key pair independently of all other key pairs in the SLH-DSA
+hypertree.
+
+A SLH-DSA tree MUST NOT be used for more than 2^64 signing
+operations.
+
+The generation of private keys relies on random numbers.  The use of
+inadequate pseudo-random number generators (PRNGs) to generate these
+values can result in little or no security.  An attacker may find it
+much easier to reproduce the PRNG environment that produced the keys,
+searching the resulting small set of possibilities, rather than brute
+force searching the whole key space.  The generation of quality
+random numbers is difficult, and {{?RFC4086}} offers important guidance
+in this area.
 
-TODO Security
+When computing signatures, the same hash function SHOULD be used to
+compute the message digest of the content and the signed attributes,
+if they are present.
+
+When computing signatures, implementations SHOULD include protections
+against fault injection attacks [CMP2018].  Protections against these
+attacks include signature verification prior to releasing the
+signature value to confirm that no error injected and generating the
+signature a few times to confirm that the same signature value is
+produced each time.
```

Comment:
Here we can give reference to the new paper of my friend Markku Saarinen who suggests further recommendations and, moreover, provides some insights into side-channel resistant implementation of SLH-DSA.

Reply:
https://eprint.iacr.org/2024/367